### PR TITLE
Make Inputdriver5 an experimental Feature

### PR DESCRIPTION
**Pull request by MichaelPFrey**
Added the inputdriver to the list of experimental features.
CONFIG+=experimental as we explicitly work on an experimental feature.
Add code to stop the drivers when the features gets disabled during run time.
Hide the UI when the feature is disabled.

I think it makes sense tobe able to disable the inputdriver(s), even when the feature becomes one day stable.

### CODE REVIEW

File: src/feature.cc [hunk 1]
```diff
@@ -27,7 +27,7 @@ const Feature Feature::ExperimentalForCExpression("lc-for-c", "Enable C-style <c
 const Feature Feature::ExperimentalAmfImport("amf-import", "Enable AMF import.");
 const Feature Feature::ExperimentalSvgImport("svg-import", "Enable SVG import.");
 const Feature Feature::ExperimentalCustomizer("customizer", "Enable Customizer");
-
+const Feature Feature::ExperimentalInputDriver("input-drive", "Enable input drivers (requires restart)");
```
Comment (kintel) [inline src/feature.cc]: Should this be "input-driver" ?

File: src/input/InputDriverManager.cc [hunk 2]
```diff
@@ -129,6 +129,18 @@ std::string InputDriverManager::listDrivers()
     return stream.str();
 }
 
+void InputDriverManager::closeDrivers()
+{
+	stopRequest = true;
+	timer->stop();
+	InputEventMapper::instance()->stop();
+
+    for (drivers_t::iterator it = drivers.begin();it != drivers.end();it++) {
```
Comment (kintel) [inline src/input/InputDriverManager.cc]: C++11 range iterators could be used here: `for (auto driver : drivers) {`.

Also: indentation mismatch (in multiple places in this PR)

File: openscad.pro [hunk 3]
```diff
@@ -1,3 +1,4 @@
+CONFIG+=experimental
```
Comment (kintel) [inline openscad.pro]: This shouldn't be the default

File: src/feature.cc (hunk 1 above)
Comment (MichaelPFrey) [reply src/feature.cc]: typo - thank you for catching it

File: openscad.pro (hunk 3 above)
Comment (MichaelPFrey) [reply openscad.pro]: Reverted, but I am kind of split on this. Yes, this should not/never come into the master branch, we agree on that.
On the other hand: This is a feature branch were we are working on an experimental feature.
Someone that is checking out an experimental feature branch may not know that on top of checking out the experimental branch has to explicitly add +CONFIG+=experimental
(even when knowing it, it is an extra step)
I guess the input-driver is kind of a special feature branch

File: src/input/InputDriverManager.cc (hunk 2 above)
Comment (MichaelPFrey) [reply src/input/InputDriverManager.cc]: changed for loops to c++11 for loop
indentation matched to preexisting code

File: openscad.pro (hunk 3 above)
Comment (kintel) [reply openscad.pro]: Good point. Perhaps @t-paul could comment on this as we need a mechanism for remembering to revert such things before merging to master.

**Comment by kintel** [comment]
@t-paul This is also yours - code was reviewed.

**Pull request merged**